ivadomed/inference.py

```diff
@@ -314,7 +330,7 @@ def split_classes(nib_prediction):
     pred = nib_prediction.get_fdata()
     pred_list = []
     for c in range(pred.shape[-1]):
-        class_pred = nib.Nifti1Image(pred[..., c].astype('float32'), nib_prediction.affine)
+        class_pred = nib.Nifti1Image(pred[..., c].astype('float32'), None, nib_prediction.header)
```

Comment:
need to copy to another object (to avoid pointer conflicts)
```suggestion
        class_pred = nib.Nifti1Image(pred[..., c].astype('float32'), None, nib_prediction.header.copy())
```